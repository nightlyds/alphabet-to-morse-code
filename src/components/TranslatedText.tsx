/* eslint-disable react/require-default-props */
import React, { useEffect, useState } from "react";
import { CopyToClipboard } from "react-copy-to-clipboard";

export type TranslatedTextProps = {
    inputText?: string;
};

const TranslatedText = ({ inputText = "" }: TranslatedTextProps) => {
    const [translatedText, setTranslatedText] = useState<string>("");

    useEffect(() => {
        setTranslatedText(inputText);
    }, [inputText]);
    return (
        <div className="translated-text">
            <textarea
                className="translated-text-textarea"
                disabled
                placeholder={translatedText}
            />
            <div className="translated-text-copy">
                <CopyToClipboard text={translatedText}>
                    <span className="translated-text-copy-link">Copy</span>
                </CopyToClipboard>
            </div>
        </div>
    );
};

export default TranslatedText;
