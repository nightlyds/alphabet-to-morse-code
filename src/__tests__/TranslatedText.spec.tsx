import React from "react";
import { shallow } from "enzyme";
import TranslatedText from "../components/TranslatedText";

let setUp = () => shallow(<TranslatedText />);

describe("TranslatedText tests", () => {
    describe("Snapshot tests", () => {
        let component: any;

        beforeEach(() => {
            component = setUp();
        });

        it("Basic snapshot test", () => {
            expect(component).toMatchSnapshot();
        });
    });
});
