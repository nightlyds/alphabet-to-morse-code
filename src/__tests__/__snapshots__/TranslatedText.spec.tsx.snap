// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslatedText tests Snapshot tests Basic snapshot test 1`] = `
<div
  className="translated-text"
>
  <textarea
    className="translated-text-textarea"
    disabled={true}
    placeholder=""
  />
  <div
    className="translated-text-copy"
  >
    <CopyToClipboard
      text=""
    >
      <span
        className="translated-text-copy-link"
      >
        Copy
      </span>
    </CopyToClipboard>
  </div>
</div>
`;
